### DISCUSSION TITLE
Why is snprintf (svalue, sizeof(svalue), "%f", value); not working with float?

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#16006] [Q&A] [answered]
When I use snprintf (svalue, sizeof(svalue), "%f", value); svalue contains no value.
When I use snprintf (svalue, sizeof(svalue), "%d", value); with a decimal value it works ok.

Why is %f not working with (sn)printf and how can I fill my string with the float value?

In arduino this works fine but in platformio with Tasmota I get empty strings for "%f"

**Comment by arendst** [comment, marked as answer]
To save precious code space we do not support floats with snprintf.

We made our own implementation using ext_snprintf_P()

Search the code how to use it (and watch for the mandatory ampersand &)

**Comment by jeroenst** [comment]
Ok, thanks, that works.

**Comment by jeroenst** [comment]
I tried xt_snprintf_P(txt, sizeof(txt), PSTR("%\*.\*_f"), width, precision, &fnum); but it doesn't seem to work. According to the fprint documentation the width parameter should be correct in this way. Could it be that the width parameter is not supported?

**Reply by arendst** [reply]
The dot needs to be an asterix and delete width

**Reply by jeroenst** [reply]
Sorry formatting of github did remove the asterisks. I meant: ext_snprintf_P(txt, sizeof(txt), PSTR("%\*.\*_f"), width, precision, &fnum);

**Reply by s-hadinger** [reply]
Check the documentation https://tasmota.github.io/docs/Sensor-API/#how-to-use

Float formatting accepts only one `*` parameter for the number of decimals. It is not standard but it was a compromise to keep code small enough.

**Reply by jeroenst** [reply]
Thank you for pointing out!
